integration-tests/sync-tests/sync-verifier.spec.ts

```diff
@@ -0,0 +1,96 @@
+import chai, { expect } from 'chai'
+import { Wallet, BigNumber, providers } from 'ethers'
+import { injectL2Context } from '@eth-optimism/core-utils'
+
+import { sleep, l2Provider, verifierProvider } from '../test/shared/utils'
+import { OptimismEnv } from '../test/shared/env'
+import { DockerComposeNetwork } from '../test/shared/docker-compose'
+
+describe('Syncing a verifier', () => {
+  let env: OptimismEnv
+  let wallet: Wallet
+  let verifier: DockerComposeNetwork
+
+  const sequencerProvider = injectL2Context(l2Provider)
+
+  /* Helper functions */
+
+  const waitForBatchSubmission = async (
+    totalElementsBefore: BigNumber
+  ): Promise<BigNumber> => {
+    // Wait for batch submission to happen by watching the CTC
+    let totalElementsAfter = (await env.ctc.getTotalElements()) as BigNumber
+    while (totalElementsBefore.eq(totalElementsAfter)) {
+      await sleep(500)
+      totalElementsAfter = (await env.ctc.getTotalElements()) as BigNumber
+    }
+    return totalElementsAfter
+  }
+
+  const startAndSyncVerifier = async (sequencerBlockNumber: number) => {
+    // Bring up new verifier
+    verifier = new DockerComposeNetwork(['verifier'])
+    await verifier.up({ commandOptions: ['--scale', 'verifier=1'] })
+
+    // Wait for verifier to be looping
+    let logs = await verifier.logs()
+    while (!logs.out.includes('Starting Sequencer Loop')) {
+      await sleep(500)
+      logs = await verifier.logs()
+    }
+
+    const provider = injectL2Context(verifierProvider)
+
+    // Wait until verifier has caught up to the sequencer
+    let latestVerifierBlock = (await provider.getBlock('latest')) as any
+    while (latestVerifierBlock.number < sequencerBlockNumber) {
+      await sleep(500)
+      latestVerifierBlock = (await provider.getBlock('latest')) as any
+    }
+
+    return provider.getBlock(sequencerBlockNumber)
+  }
+
+  before(async () => {
+    env = await OptimismEnv.new()
+    wallet = env.l2Wallet
+  })
+
+  describe('Basic transactions', () => {
+    afterEach(async () => {
+      await verifier.stop('verifier')
+      await verifier.rm()
+    })
+
```

Comment:
Why does `afterEach` remove the verifier but there is no `beforeEach` to bring it up? Shouldn't this be a single `before` to bring it up (decouple starting and syncing) and just start and then a single before to clean up?

Reply:
i followed the structure described in https://github.com/ethereum-optimism/optimism/issues/604 which specified spinning up the verifier after making transactions. i haven't looked into distinctions between the two approaches yet, just imagined this test as "starting with a fresh copy every time"